lib/galaxy/webapps/galaxy/api/tools.py

```diff
@@ -557,7 +554,7 @@ def _create(self, trans: GalaxyWebTransaction, payload, **kwd):
         # as a regular tool parameter we accept both.
         use_cached_job = payload.get('use_cached_job', False) or util.string_as_bool(inputs.get('use_cached_job', 'false'))
 
-        input_format = kwd.get('input_format', 'legacy')
+        input_format = payload.get('input_format', 'legacy')
```

Comment:
```suggestion
        input_format = payload.get('input_format', '21.01')
```
A bit unrelated - this wasn't discussed in the original PR for some reason, but maybe it makes sense that '21.01' is actually the default value in Galaxy version 21.01?

Reply:
I guess the reason to keep `legacy` is default is that Galaxy API is generally very conservative in trying not to break backward-compatibility.

Reply:
That and the frontend doesn't submit in this format.